server/service/mdm.go

```diff
@@ -2618,130 +2619,140 @@ func (svc *Service) UploadMDMAppleVPPToken(ctx context.Context, token io.ReadSee
 	}
 
 	act := fleet.ActivityEnabledVPP{}
 	if err := svc.NewActivity(ctx, authz.UserFromContext(ctx), act); err != nil {
-		return ctxerr.Wrap(ctx, err, "create activity for upload VPP token")
+		return nil, ctxerr.Wrap(ctx, err, "create activity for upload VPP token")
 	}
 
-	return nil
+	return tok, nil
 }
 
-////////////////////////////////////////////////////////////////////////////////
-// GET /vpp
-////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////
+// PATCH /api/_version_/fleet/vpp_tokens/%d/teams //
+////////////////////////////////////////////////////
 
-type getMDMAppleVPPTokenRequest struct{}
+type patchVPPTokensTeamsRequest struct {
+	ID      uint   `url:"id"`
+	TeamIDs []uint `json:"team_id"`
+}
 
-type getMDMAppleVPPTokenResponse struct {
-	*fleet.VPPTokenInfo
+type patchVPPTokensTeamsResponse struct {
 	Err error `json:"error,omitempty"`
 }
 
-func (r getMDMAppleVPPTokenResponse) error() error {
-	return r.Err
-}
+func (r patchVPPTokensTeamsResponse) error() error { return r.Err }
 
-func getMDMAppleVPPTokenEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (errorer, error) {
-	vpp, err := svc.GetMDMAppleVPPToken(ctx)
-	if err != nil {
-		return &getMDMAppleVPPTokenResponse{Err: err}, nil
-	}
+func patchVPPTokensTeams(ctx context.Context, request any, svc fleet.Service) (errorer, error) {
+	req := request.(*patchVPPTokensTeamsRequest)
 
-	return &getMDMAppleVPPTokenResponse{VPPTokenInfo: vpp}, nil
+	if err := svc.UpdateVPPTokenTeams(ctx, req.ID, req.TeamIDs); err != nil {
+		return patchVPPTokensTeamsResponse{Err: err}, nil
+	}
+	return patchVPPTokensTeamsResponse{}, nil
 }
 
-func (svc *Service) GetMDMAppleVPPToken(ctx context.Context) (*fleet.VPPTokenInfo, error) {
-	if err := svc.authz.Authorize(ctx, &fleet.AppleCSR{}, fleet.ActionRead); err != nil {
-		return nil, err
+func (svc *Service) UpdateVPPTokenTeams(ctx context.Context, tokenID uint, teamIDs []uint) error {
+	if err := svc.authz.Authorize(ctx, &fleet.AppleCSR{}, fleet.ActionWrite); err != nil {
+		return err
 	}
 
-	assetMap, err := svc.ds.GetAllMDMConfigAssetsByName(ctx, []fleet.MDMAssetName{fleet.MDMAssetVPPTokenDeprecated})
-	if err != nil {
-		return nil, ctxerr.Wrap(ctx, err, "get mdm config assets by name VPP token")
+	if teamIDs != nil && len(teamIDs) > 1 {
+		return ctxerr.Wrap(ctx, fleet.NewUserMessageError(errors.New("A VPP token can only belong to one team"), http.StatusBadRequest))
 	}
 
-	var tokenData fleet.VPPTokenData
-	if err := json.Unmarshal(assetMap[fleet.MDMAssetVPPTokenDeprecated].Value, &tokenData); err != nil {
-		return nil, ctxerr.Wrap(ctx, err, "unmarshaling VPP token data")
+	var nullTeam fleet.NullTeamType
+	var teamID *uint
+	if teamIDs == nil {
+		nullTeam = fleet.NullTeamNoTeam
+	} else if len(teamIDs) == 0 {
+		nullTeam = fleet.NullTeamAllTeams
+	} else if teamIDs[0] == 0 {
+		nullTeam = fleet.NullTeamNone
```

Comment:
And this would be "no team"?

Reply:
Looks like I got the no team and none teams cases switched up